MacrobenchmarkSample/app/src/main/java/com/example/macrobenchmark/target/util/Data.kt

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2022 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.macrobenchmark.target.util
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import java.util.Random
+import kotlin.coroutines.CoroutineContext
+
+abstract class Data {
+    abstract suspend fun isReady(): Boolean
+}
+
+class RandomData : Data() {
+    private val random = Random()
+    override suspend fun isReady(): Boolean {
+        CoroutineScope(Dispatchers.Default).launch {
```

Comment:
We don't need to relaunch coroutine, just switch dispatcher with `withContext` 
```suggestion
        withContext(Dispatchers.Default) { 
```

Reply:
obsolete